src/AWS.Deploy.CLI/Commands/DeployCommand.cs

```diff
@@ -458,8 +461,7 @@ private void ConfigureDeploymentFromConfigFile(Recommendation recommendation, Us
             }
 
             var validatorFailedResults =
-                        recommendation.Recipe
-                            .BuildValidators()
+                _validatorFactory.BuildValidators(recommendation.Recipe)
                             .Select(validator => validator.Validate(recommendation, _session, _optionSettingHandler))

```

Comment:
could you inject `_optionSettingHandler` as a dependency of `_validatorFactory`? that way we wouldn't need to update the `validator.Validate` signature every time we need a new service added?

Reply:
Updated both existing recipe validators with a `IOptionSettingHandler` in their constructor, which is injected in the new `ValidatorFactory`.  Removed `_optionSettingHandler` from the `Validate` calls.